bindings/java/src/org/sleuthkit/datamodel/Report.java

```diff
@@ -18,49 +18,74 @@
  */
 package org.sleuthkit.datamodel;
 
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import static java.nio.file.StandardOpenOption.READ;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 /**
- * This is a data transfer object (DTO) class that models reports.
+ * This is a class that models reports.
  */
-public class Report {
+public class Report implements Content {
 
 	static long ID_NOT_SET = -1;
 	private long id = ID_NOT_SET;
-	private final String path;
+	private final Path path;
 	private final long createdTime;
 	private final String sourceModuleName;
 	private final String reportName;
+	private final Content source; // The source file used to produce the report, if available
+	private final SleuthkitCase db; // A reference to the database instance.
+	private FileChannel fileChannel; // Used to read report content.
+
+	private static final Logger LOGGER = Logger.getLogger(Report.class.getName());
 
 	/**
-	 * Construct a data transfer object for a row in the reports table.
+	 * Create a Report instance.
 	 *
 	 * @param id          Primary key from associated row in the case database.
 	 * @param path        Absolute path to report.
 	 * @param createdTime Created time of report (in UNIX epoch time).
 	 * @param reportName  May be empty
+	 * @param source	  The source from which the Report was created, if
```

Comment:
I think that it would not be crazy to require a report name. In fact it would be good for the output of the collections search, where the the Source Name for reports is the report name. All we need to change is the comment. I am not sure why I allowed the empty string.